### ISSUE TITLE
Feature Idea: Manual sorting board views

### ISSUE THREAD

**Issue by Johennes** [mattermost/focalboard#299]
**Summary:**

Currently the views in a board are sorted alphabetically in the left pane. It would be nice if the views could be sorted manually through drag and drop. Particularly when using emoji as the first character in view names, the alphabetical sorting tends to produce non-intuitive results.

![Screenshot 2021-04-20 at 08 59 12](https://user-images.githubusercontent.com/1137962/115351857-fe72fa80-a1b6-11eb-8d6b-d490481cf2e2.png)

**How important this is to me and why:**

Importance: Low

Low importance because I'm not using terribly many views (yet) so it's a minor inconvenience only.

Use cases:

1. I'm using emojis as the first character in view names which breaks intuitive alphabetical sorting
2. I want certain views to appear at the top / bottom of the list because I use them more / less frequently

**Comment by chenilim** [comment]
Thanks @Johennes! Yes, we do intend to support manual ordering of the sidebar, perhaps with additional organization. If anyone has things they would like to see, or examples of good implementations of this, please add links. Much appreciated!

**Comment by Johennes** [comment]
An easy fix for my particular issue could also be to strip leading emoji from the view names before applying the alphabetical sorting.

**Comment by chenilim** [comment]
That might be a good short-term fix @Johennes. If someone wants to do that first, this ticket is still up for grabs! :)

**Comment by Johennes** [comment]
I am someone so I gave this a try in #341. 😅

**Comment by chenilim** [comment]
@asaadmahmood, assigning this to you to track as part of the sidebar re-design. Thanks.

**Comment by nqtronix** [comment]
I wanted to re-order the entries on the left sidebar, so I tried in order:

1. drag & drop
2. right click
3. 3-dot-menu
4. opening an entry (re-naming an entry requires this)
5. re-naming entries as work around\*
6. reading the help menu (which doesn't mention the order)
7. searching on github issues

IMHO the main problem is that the user (me) may expect that manual sorting *should* work, and may assume he's just too stupid to find it. After all, some common features are not where they usually are (no re-name on right click/ 3-dot-menu).

Any yes, I *did* try rename the entries with a leading number to force sorting, but it didn't work either. After reading this issue I assume not only leading emojis are ignored, but also leading numbers? I've not experience with typescript, so I'd be great if someone can verify and fix this (=re-enable sort by leading numbers).

I also think the user guide should include a hint, that sorting happens automatically & is based on the name. I could edit this in myself, but because the user guide is a bit outdated anyway, it should be updated in general. If you want me to do that, please let me know.

**Comment by nqtronix** [comment]
>After reading this issue I assume not only leading emojis are ignored, but also leading numbers?

A quick and dirty fix is to lead with a space, then type the number.

**Comment by asaadmahmood** [comment]
@nqtronix Agreed, the experience can be considerably improved. We're looking into this along with other sidebar improvements.
